### PR TITLE
Display Users

### PR DESCRIPTION
**Pull request by SSalekin**

### CODE REVIEW

File: app/assets/stylesheets/registrations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the registrations controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (MuhammadTamzid) [inline app/assets/stylesheets/registrations.scss]: remove unused file

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (MuhammadTamzid) [inline app/assets/stylesheets/users.scss]: remove unused file

File: app/assets/javascripts/registrations.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (sujoydatta) [inline app/assets/javascripts/registrations.coffee]: remove unused files

File: app/assets/javascripts/users.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (sujoydatta) [inline app/assets/javascripts/users.coffee]: above

File: app/controllers/registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class RegistrationsController < Devise::RegistrationsController
+  private
+
+  def sign_up_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
```
Comment (MuhammadTamzid) [inline app/controllers/registrations_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.paginate page: params[:page], per_page: 20
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (sujoydatta) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/registrations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  def account_update_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation, :current_password)
```
Comment (MuhammadTamzid) [inline app/controllers/registrations_controller.rb]: remove ()

File: app/helpers/registrations_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module RegistrationsHelper
+end
```
Comment (sujoydatta) [inline app/helpers/registrations_helper.rb]: remove

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: use before action

File: app/helpers/users_helper.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  def gravatar_for(user, options = { size: 160 })
```
Comment (MuhammadTamzid) [inline app/helpers/users_helper.rb]: remove ()

File: config/routes.rb [hunk 10]
```diff
@@ -10,7 +10,7 @@
       resources :categories
     end
 
-    devise_for :users
+    devise_for :users, :controllers => { registrations: 'registrations' }
```
Comment (sujoydatta) [inline config/routes.rb]: controllers: { registrations: "registrations" }

File: app/views/devise/registrations/new.html.erb [hunk 11]
```diff
@@ -4,8 +4,13 @@
   <%= devise_error_messages! %>
 
   <div class="field">
+    <%= f.label :name %>
+    <%= f.text_field :name, class: 'form-control' %>
```
Comment (MuhammadTamzid) [inline app/views/devise/registrations/new.html.erb]: double quote

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+  def index
+    @users = User.paginate page: params[:page], per_page: 20
+  end
+
+  def show
+  end
+
+  private
+  # Before filters
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove all remark

File: app/helpers/users_helper.rb [hunk 13]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  def gravatar_for user, options = { size: 160 }
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: convention, should
```
{size: 160}
```

File: app/helpers/users_helper.rb [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+    gravatar_id = Digest::MD5::hexdigest user.email.downcase
+    size = options[:size]
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
+    image_tag(gravatar_url, alt: user.name, class: "gravatar")
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/views/users/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,19 @@
+<h1><%= t(:all_users) %></h1>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: remove `()`

File: app/views/users/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+<table class="table">
+  <thead>
+    <th><%= t(:name) %></th>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: remove `()`

File: app/views/users/index.html.erb [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+    <th><%= t(:email) %></th>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: remove `()`

File: app/views/users/show.html.erb [hunk 18]
```diff
@@ -0,0 +1,18 @@
+<% provide(:title, @user.name) %>
```
Comment (dieunb) [inline app/views/users/show.html.erb]: remove `()`

File: app/views/users/show.html.erb [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+<h1><%= t(:user_profile) %></h1>
```
Comment (dieunb) [inline app/views/users/show.html.erb]: remove `()`

File: app/views/users/show.html.erb [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+<div class="row">
+  <div class="panel panel-default">
+    <div class="panel-heading">
+      <div class="pull-right">
+      </div>
+    </div>
+    <div class="panel-body">
+      <%= gravatar_for @user %>
+      <label><%= t(:name) %></label>
```
Comment (dieunb) [inline app/views/users/show.html.erb]: remove `()`

File: app/views/users/show.html.erb [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+      <p class="form-control profile_align"><%= @user.name %></p>
+      <label><%= t(:email) %></label>
```
Comment (dieunb) [inline app/views/users/show.html.erb]: remove `()`

File: config/routes.rb [hunk 22]
```diff
@@ -10,7 +10,7 @@
       resources :categories
     end
 
-    devise_for :users
+    devise_for :users, :controllers => {registrations: "registrations"}
```
Comment (dieunb) [inline config/routes.rb]: convention, should
```
controllers: {registrations: "registrations"}
```

**Pull request merged**